client/icons/icon-components/zoomIcons.jsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
```

Comment:
Could we just put the icons as plain SVGs along with all our other custom icons? I don't think we need to devote a whole separate component for icons when we already have icons being used in plenty of other spaces. Titles, etc. can be just applied on the button itself instead of embedded in the SVG.

Reply:
I'll mess with it again.  I was having trouble with getting the color of the icons to be adaptable to whatever color their "context" was (changing the `color` property on a container div should also change the color of the icon).  As a component, I can use `currentColor`.  

For example in the snippet bar, with some custom icons, you can see that the color of the icon remains 'black' even if i specifically set those icons to `color: red`:

<img width="335" alt="image" src="https://github.com/user-attachments/assets/493d21df-efd6-4771-932d-efe398da74f8">

Reply:
Check out this reply. Apparently this is how FontAwesome gets its SVGs to use the `color`. We should be able to do this as well:

https://stackoverflow.com/a/65147574

Reply:
Yeah, i think that is basically what I have done.  The method in that SO answer is to use `currentColor` in the SVG itself...but it requires you to inline the SVG in the HTML (you can't use `<img>`), so rather than a big SVG string in the middle of the file I am dropping it into a component as a separate file (using SVGR to write the component, as noted in the answer comments).

Reply:
Ok, this is gonna sound a little crazy. What if we make our custom icons masks, instead of displaying it as the `content` property. For example, replace the `customIcons.less` with this:

```
.fac {
  display          : inline-block;
  background-color : currentColor;
  mask-size        : contain;
  mask-repeat      : no-repeat;
  mask-position    : center;
}
.position-top-left {
  mask-image: url('../icons/position-top-left.svg');
}
.position-top-right {
  mask-image: url('../icons/position-top-right.svg');
}
.position-bottom-left {
  mask-image: url('../icons/position-bottom-left.svg');
}
.position-bottom-right {
  mask-image: url('../icons/position-bottom-right.svg');
}
.position-top {
  mask-image: url('../icons/position-top.svg');
}
.position-right {
  mask-image: url('../icons/position-right.svg');
}
.position-bottom {
  mask-image: url('../icons/position-bottom.svg');
}
.position-left {
  mask-image: url('../icons/position-left.svg');
}
.mask-edge {
  mask-image: url('../icons/mask-edge.svg');
}
.mask-corner {
  mask-image: url('../icons/mask-corner.svg');
}
.mask-center {
  mask-image: url('../icons/mask-center.svg');
}
.book-front-cover {
	mask-image: url('../icons/book-front-cover.svg');
}
.book-back-cover {
  mask-image: url('../icons/book-back-cover.svg');
}
.book-inside-cover {
  mask-image: url('../icons/book-inside-cover.svg');
}
.book-part-cover {
  mask-image: url('../icons/book-part-cover.svg');
}
.davek {
  mask-image: url('../icons/Davek.svg');
}
.rellanic {
  mask-image: url('../icons/Rellanic.svg');
}
.iokharic {
  mask-image: url('../icons/Iokharic.svg');
}
```

And voila! All our custom icons can be recolored:

![image](https://github.com/user-attachments/assets/f1383d88-eb30-4533-a505-afc236fd00dd)

Reply:
Okay, done and done.